### PR TITLE
Add --include-private flag to show private fields and types

### PR DESCRIPTION
**Pull request by MoritzBrueckner**
Issue https://github.com/HaxeFoundation/dox/issues/39 is open since 2014. This pull request adds a `--private` argument that includes private fields in the generated output.

### CODE REVIEW

File: src/dox/Dox.hx [hunk 1]
```diff
@@ -65,6 +65,9 @@ class Dox {
 			@doc("Disable markdown rendering")
 			["--no-markdown"] => function() cfg.useMarkdown = false,
 
+			@doc("Show private fields")
```
Comment (Gama11) [inline src/dox/Dox.hx]: Is this accurate? Seems like it applies to types too, not just fields.

Comment (MoritzBrueckner) [reply src/dox/Dox.hx]: That's true. Should I change the description or remove the inclusion of private types?

File: src/dox/Dox.hx [hunk 2]
```diff
@@ -65,6 +65,9 @@ class Dox {
...
+			["--private"] => function() cfg.showPrivate = true,
```
Comment (Gama11) [inline src/dox/Dox.hx]: Maybe `--include-private` would be a better name, since we already have `--include`?

Comment (MoritzBrueckner) [reply src/dox/Dox.hx]: Sounds good

File: src/dox/Dox.hx (hunk 1 above)
Comment (Gama11) [reply src/dox/Dox.hx]: I'd say update the description, if you're going to show private fields you might as well show private types too.

Comment (MoritzBrueckner) [reply src/dox/Dox.hx]: All right, will push another commit in a moment

**Comment by Simn** [comment]
Looks good to me, feel free to merge if CI agrees.

**Comment by Gama11** [comment]
@MoritzBrueckner Thanks!

**Pull request merged**